basis/ui/backend/gtk/gtk.factor

```diff
@@ -219,26 +219,31 @@ icon-data [ default-icon-data ] initialize
 ! Window state events
 
 : on-expose ( win event user-data -- ? )
-    2drop window relayout t ;
+    2drop gtk_widget_get_toplevel window relayout t ;
 
-: on-configure ( win event user-data -- ? )
-    drop swap window dup active?>> [
-        swap GdkEventConfigure memory>struct
-        [ event-loc >>window-loc ] [ event-dim >>dim ] bi
-        relayout-1
-    ] [ 2drop ] if f ;
+: on-configure ( window event user-data -- ? )
+    drop swap dup gtk_widget_get_toplevel [ = ] keep window dup active?>> [
+        swap [ swap GdkEventConfigure memory>struct ] dip
+        [ event-loc >>window-loc drop ]
+        [ event-dim >>dim relayout-1 ] if
+    ] [ 3drop ] if f ;
 
 : on-map ( win event user-data -- ? )
     2drop window t >>active? drop t ;
 
 : on-delete ( win event user-data -- ? )
     2drop window ungraft t ;
 
+: connect-configure-signal ( winhandle -- )
```

Comment:
Maybe we should hook up two different words to the two different elements.

``
: on-configure-window ( window -- )

: on-configure-drawable ( drawable -- )
```

so you don't have to do the ``gtk_widget_get_toplevel [ = ] keep`` ?